master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java

```diff
@@ -213,14 +213,16 @@ public void updateWorkerHeartbeatMeta(
 
     appDiskUsageMetric.update(estimatedAppDiskUsage);
     // If using HDFSONLY mode, workers with empty disks should not be put into excluded worker list.
-    long healthyDiskNum =
-        disks.values().stream().filter(s -> s.status().equals(DiskStatus.HEALTHY)).count();
+    long unhealthyDiskNum =
+        disks.values().stream().filter(s -> !s.status().equals(DiskStatus.HEALTHY)).count();
+    boolean exceed =
+        unhealthyDiskNum * 1.0 / disks.size()
+            >= conf.masterExcludeWorkerUnhealthyDiskRatioThreshold();
```

Comment:
`conf.masterExcludeWorkerUnhealthyDiskRatioThreshold()` can be kept as a viriable.